realm/src/main/java/io/realm/Realm.java

```diff
@@ -813,17 +809,21 @@ private static void initializeRealm(Realm realm) {
         }
         checkHasPrimaryKey(clazz);
 
-        JsonReader reader = new JsonReader(new InputStreamReader(in, "UTF-8"));
+        // As we need the primary key value we have to first parse the entire input stream as in the general
```

Comment:
Yes it does. Which is why there would be totally valid points for removing this method completely. Right now it is there to provide symmetry with the other methods, and I couldn't make up my mind if we should just remove it.

What do you think @kneth @bmunkholm ?